tests/v2/Controller/DocumentController/MoveToFolderTest.php

```diff
@@ -59,31 +59,46 @@ public function provideTestRoute(): ?\Generator
+        $user = $this->getTestUserFromDb(BeneficiaryFixture::BENEFICIARY_MAIL);
         $clientTest->loginUser($user);
 
         $testedBeneficiary = $user->getSubjectBeneficiaire();
         $randomBeneficiary = BeneficiaireFactory::findByEmail(BeneficiaryFixture::BENEFICIARY_MAIL_SETTINGS)->object();
 
         $testedBeneficiaryDocument = DocumentFactory::createOne(['beneficiaire' => $testedBeneficiary, 'bPrive' => false])->object();
-        $testedBeneficiaryFolder = FolderFactory::createOne(['beneficiaire' => $testedBeneficiary, 'bPrive' => true])->object();
         $randomBeneficiaryFolder = FolderFactory::createOne(['beneficiaire' => $randomBeneficiary])->object();
 
         // Tested beneficiary tries to move document inside random beneficiarie's folder
         $clientTest->request('GET', sprintf(self::URL, $testedBeneficiaryDocument->getId(), $randomBeneficiaryFolder->getId()));
         self::assertResponseStatusCodeSame(403);
+    }
+
+    public function provideTestMoveToFolder(): ?\Generator
+    {
+        yield 'Shared document should be hydrated with parent visibility' => [false];
+        yield 'Private document should be hydrated with parent visibility' => [true];
+    }
+
+    /** @dataProvider provideTestMoveToFolder */
+    public function testMoveToFolder(bool $isPrivate): void
+    {
+        $clientTest = static::createClient();
+        $user = $this->getTestUserFromDb(BeneficiaryFixture::BENEFICIARY_MAIL);
+        $clientTest->loginUser($user);
+        $beneficiary = $user->getSubjectBeneficiaire();
+
+        // Document and destination folder have different visibility
+        $document = DocumentFactory::findOrCreate(['beneficiaire' => $beneficiary, 'bPrive' => $isPrivate])->object();
+        $folder = FolderFactory::findOrCreate(['beneficiaire' => $beneficiary, 'bPrive' => !$isPrivate])->object();
 
-        // Then move in a folder that belongs to him
-        $clientTest->request('GET', sprintf(self::URL, $testedBeneficiaryDocument->getId(), $testedBeneficiaryFolder->getId()));
-        $testedBeneficiaryDocument = DocumentFactory::find(['id' => $testedBeneficiaryDocument->getId()]);
-        $testedBeneficiaryFolder = FolderFactory::find(['id' => $testedBeneficiaryFolder->getId()]);
+        $clientTest->request('GET', sprintf(self::URL, $document->getId(), $folder->getId()));
+        $document = DocumentFactory::find($document)->object();
+        $folder = FolderFactory::find($folder)->object();
 
-        self::assertEquals($testedBeneficiaryDocument->object()->getDossier()->getId(), $testedBeneficiaryFolder->object()->getId());
-        self::assertTrue($testedBeneficiaryFolder->object()->getbPrive());
-        $testedBeneficiaryDocument->remove();
-        $testedBeneficiaryFolder->remove();
+        self::assertEquals($document->getDossier()->getId(), $folder->getId());
+        self::assertEquals($document->getBprive(), $folder->getBPrive());
```

Comment:
It's not that important but first argument of assertEquals should be the expected result. So, in this case `$folder->getBPrive()` cause we expect document visibility to be the same as its parent folder.